src/quick-lint-js/fe/parse-expression.cpp

```diff
@@ -2065,6 +2065,19 @@ Expression* Parser::parse_expression_remainder(Parse_Visitor_Base& v,
     break;
 
   case Token_Type::left_curly: {
+    auto has_comma_operator =
+        [](Binary_Expression_Builder* binary_builder) -> bool {
```

Comment:
Must fix: GCC complains about shadowing here: https://github.com/quick-lint/quick-lint-js/actions/runs/6199142690/job/16831061448?pr=1083

I think we should capture instead of using a parameter:
```suggestion
    auto has_comma_operator = [&binary_builder]() -> bool {
```